atom/browser/api/atom_api_system_preferences_mac.mm

```diff
@@ -144,6 +144,84 @@
   }
 }
 
+void SystemPreferences::SetUserDefault(const std::string& name,
+                                       const std::string& type,
+                                       mate::Arguments* args) {
+  NSUserDefaults* defaults = [NSUserDefaults standardUserDefaults];
+  NSString* key = base::SysUTF8ToNSString(name);
+  if (type == "string") {
+    std::string value;
+    if (!args->GetNext(&value)) {
+      args->ThrowError();
```

Comment:
What do you think about adding error messages for each type error here? Something like "Unable to convert value to string".